modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -1121,3 +1144,53 @@ def test_reindex_preserve_dtypes(kwargs):
 
     reindexed_df = df.reindex(**kwargs)
     assert reindexed_df._query_compiler._modin_frame.has_materialized_dtypes
+
+
+def test_skip_set_columns():
+    """
+    Verifies that the mechanism of skipping the actual ``._set_columns()`` call in case
+    the new columns are identical to the previous ones works properly.
+
+    In this test, we rely on the ``modin_frame._deferred_column`` attribute.
+    The new indices propagation is done lazily, and the ``deferred_column`` attribute
+    indicates whether there's a new indices propagation pending.
+    """
+    df = pd.DataFrame({"col1": [1, 2, 3], "col2": [3, 4, 5]})
+    df.columns = ["col1", "col10"]
+    # Verifies that the new columns were successfully set in case they're actually new
+    assert df._query_compiler._modin_frame._deferred_column
+    assert np.all(df.columns.values == ["col1", "col10"])
+
+    df = pd.DataFrame({"col1": [1, 2, 3], "col2": [3, 4, 5]})
+    df.columns = ["col1", "col2"]
+    # Verifies that the new columns weren't set if they're equal to the previous ones
+    assert not df._query_compiler._modin_frame._deferred_column
+
+    df = pd.DataFrame({"col1": [1, 2, 3], "col2": [3, 4, 5]})
+    df.columns = pandas.Index(["col1", "col2"], name="new name")
+    # Verifies that the new columns weren't set if they're equal to the previous ones
```

Comment:
Is the comment outdated?

Reply:
right, it was outdated.

Changed it to "Verifies that the new columns were successfully set in case they's new metadata"